# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GC.cpp
# Commit: 50bc593bee02
# Full Hash: 50bc593bee02c47ee068d36ba11754ee1a257a61
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2019-08-19 16:09:32
# Description:
#   Bug 1574339 - Use a default GC invocation kind if embedding callbacks finish the current GC as we're tring to start a slice r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D42331
# ==============================================================================

diff -r 5448f3aaa09a -r 50bc593bee02 js/src/gc/GC.cpp
--- a/js/src/gc/GC.cpp	Thu Aug 15 20:02:33 2019 +0000
+++ b/js/src/gc/GC.cpp	Sat Aug 17 00:50:18 2019 +0000
@@ -7086,7 +7086,7 @@
   switch (incrementalState) {
     case State::NotActive:
       incMajorGcNumber();
-      invocationKind = gckind.value();
+      invocationKind = gckind.valueOr(GC_NORMAL);
       initialReason = reason;
       cleanUpEverything = ShouldCleanUpEverything(reason, invocationKind);
       sweepOnBackgroundThread = ShouldSweepOnBackgroundThread(reason);
